File: src/py/postgres_serial.py
from gaussian import gaussianCDF
from psycopg2 import connect

try:
  conn = connect(
    host="localhost",
    user="root",
    password="root123",
    database="gaussian",
  )
  cur = conn.cursor()
  cur.execute('''DROP TABLE IF EXISTS py_serial; CREATE TABLE py_serial(id SERIAL PRIMARY KEY,
  z_score FLOAT NOT NULL, cumulative_distribution FLOAT NOT NULL);''')

  for i in range(-500, 501):
    x = i / 100
    prob = gaussianCDF(0, 1, x)
    cur.execute(
      f'INSERT INTO py_serial(z_score, cumulative_distribution) VALUES ({x:.2f}, {prob:.6f})')

  conn.commit()
  conn.close()
except Exception as e:
	print(e)

File: src/java/Postgres_serial.java
import java.sql.*;
import java.util.Locale;

public class Postgres_serial {
  public static void main(String[] args) {
    Connection conn;
    Statement stmt;
    String query = "";
    int i;
    double x, prob;
    Locale.setDefault(new Locale("en", "US"));
    try {
      conn = DriverManager.getConnection("jdbc:postgresql://localhost/gaussian", "root", "root123");
      stmt = conn.createStatement();
      stmt.executeUpdate("DROP TABLE IF EXISTS java_serial; CREATE TABLE java_serial(id SERIAL " +
      "PRIMARY KEY, z_score REAL NOT NULL, cumulative_distribution REAL NOT NULL)");
      for (i = -500; i <= 500; i++) {
        x = (double) i / 100.0;
        prob = Gaussian.gaussianCDF(0, 1, x);
        query = String.format("INSERT INTO java_serial(z_score, cumulative_distribution) VALUES (%.2f, %f);", x, prob);
        stmt.executeUpdate(query);
      }
    } catch (Exception e) {
      System.err.println(e.getMessage());
    }
  }
}
